lib/Index/IndexRecord.cpp

```diff
@@ -471,6 +666,16 @@ static void addModuleDependencies(ArrayRef<ImportedModule> imports,
             // the underlying module instead.
             if (auto *declaring = mod->getDeclaringModuleIfCrossImportOverlay())
               moduleName = declaring->getNameStr();
+
+            // Emit symbolic interface files for any re-exported Clang modules
+            // from this Swift module.
+            if (mod->getASTContext().LangOpts.EnableCXXInterop) {
+              SmallVector<ImportedModule, 4> imports;
+              mod->getImportedModules(imports,
+                                      ModuleDecl::ImportFilterKind::Exported);
+              emitTransitiveClangSymbolicInterfacesForSwiftModuleImports(
```

Comment:
`emitDataForSwiftSerializedModule` recursively calls `addModuleDependencies`, will this cause some duplication (which would be ignored because of the up to date check)? It'd be nice to separate the walk of modules from the actual emission more, ie. have `addModuleDependencies` do a transitive walk and then the `emit*` only do the actual emitting.

Reply:
That's a good idea. I'm going to do that as follow-up though if you don't mind.

Reply:
Sure 👍. I've just been looking into this more (unrelated) and I'm not sure it's actually a good idea though.

We need to do the walk regardless to add the module dependencies into the unit. But we only do the emit for system modules. Ideally the emit for system modules would also return the path for the file rather than hoping we pick the correct one. Pulling it apart would make that impossible.

Reply:
It's maybe not that bad though, it is probably still worth splitting them apart TBH.